### PR TITLE
Tutorial for Helix akin to `vimtutor`

### PR DESCRIPTION
**Pull request by Omnikar**
Addresses #506.

Most of what I have so far is practically the same as Vimtutor, but it will eventually diverge once I reach multiple Helix-specific topics such as multiple selections. I will need to implement actually opening the tutorial using `hx --tutor` and/or `:tutor` from within the editor.

- [ ] explain extend modifier
- [ ] multi selections

### CODE REVIEW

**Comment by Omnikar** [comment]
I suppose that in order to actually incorporate the file into the binary, I should store a `static` somewhere that does an `include_str!` of the file. Where in the code should I put the `static`?

**Comment by Omnikar** [comment]
Should it create a new file with the tutorial in it and then open it in the editor, or just load the tutorial into a buffer without a filename? It seems that in order to do the latter I'd have to create a method on `Editor` in `helix-view/src/editor.rs` that creates a new buffer with some specified text, since `open` requires a file and `new_file` creates an empty buffer.

**Comment by archseer** [comment]
What if you put the file in the runtime folder? Then you'd be able to load it via runtime_dir()

**Comment by Omnikar** [comment]
Ah, yeah, I should do that. But my question still stands, as the user can't be allowed to directly access the file since the tutorial involves editing it.

**Comment by archseer** [comment]
You could copy it into `tmp`, but setting file permissions to read-only should be enough too. That should already be solved for us in packaged releases since install paths are usually read only.

**Comment by Omnikar** [comment]
I think you might have misread. I'm not worrying about the user editing the file when they're not supposed to, the tutorial actually has the user edit the buffer in order to help with teaching, I just want to make sure that any edits they make won't be able to be saved to the original file.

**Comment by archseer** [comment]
Yeah, if the file is chmodded to read-only they'll be able to edit it in helix but unable to change any of the changes to disk. That way the original file stays intact.

**Comment by Omnikar** [comment]
Ah, okay.

**Comment by Omnikar** [comment]
But, I feel like the file being readonly isn't enough to prevent it from being accidentally edited. Especially if someone installs from source, they'd have to chmod it as readonly themself. I'd rather leave the file in the runtime folder, but instead of opening it normally, load it into a buffer without linking it to the filename.

File: docs/tutor.txt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+   w - Move forward to before the beginning of the next word.
+   e - Move forward to the end of the current word.
+   b - Move backward to the beginning of the current word.
+
+ The w,e,b motions also have counterparts - W,E,B - which
+ traverse WORDS instead of words. WORDS are only separated by
+ whitespace, whereas words can be separated by other characters
+ in addition to whitespace.
+
+ All of these motions select the text they traverse.
```
Comment (pickfire) [inline docs/tutor.txt:226]: I think we should have a section discussing the multiple selections at least.

Comment (Omnikar) [reply docs/tutor.txt:226]: I am planning to add a section introducing multiple selections; the tutorial is not complete yet.

**Comment by cessen** [comment]
So far this looks good to me.  The text is really clear, and seems like a good introduction to the basics.

**Comment by archseer** [comment]
Should we merge this as is? I think it's a good start and we can add more sections in a follow up.

**Comment by Omnikar** [comment]
> Should we merge this as is? I think it's a good start and we can add more sections in a follow up.

If we decide to do that I'd probably want to add a note at the end that the tutorial is a WIP.

**Comment by pickfire** [comment]
> If we decide to do that I'd probably want to add a note at the end that the tutorial is a WIP.

We can add a TODO which is better.

**Review by pickfire — APPROVED** [review]
Good work. A good start and could be improved later on, at least selections part should be there later on.

**Pull request merged**